tough/src/lib.rs

```diff
@@ -328,14 +335,22 @@ impl<'a, T: Transport> Repository<'a, T> {
         //   HASH is one of the hashes of the targets file listed in the targets metadata file
         //   found earlier in step 4. In either case, the client MUST write the file to
         //   non-volatile storage as FILENAME.EXT.
-        Ok(
-            if let Some(target) = self.targets.signed.targets.get(name) {
-                let (sha256, file) = self.target_digest_and_filename(target, name);
-                Some(self.fetch_target(target, &sha256, file.as_str())?)
-            } else {
-                None
-            },
-        )
+        Ok(if let Ok(target) = self.targets.signed.find_target(name) {
+            let (sha256, file) = self.target_digest_and_filename(target, name);
+            Some(self.fetch_target(target, &sha256, file.as_str())?)
+        } else {
+            None
+        })
+    }
+
+    pub fn role(&self, name: &str) -> Result<&DelegatedRole> {
+        Ok(self
+            .targets
+            .signed
+            .delegated_role(name)
+            .context(error::DelegateNotFound {
+                name: name.to_string(),
```

Comment:
I'd prefer it if this method returned `Option<_>` and we returned None here instead of erroring. This is similar to other "get thing by name" methods like [`HashMap::get`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get).